taxi/tests/test_views.py

```diff
@@ -0,0 +1,98 @@
+from django.contrib.auth import get_user_model
+from django.test import TestCase, Client
+from django.urls import reverse
+
+from taxi.models import Manufacturer, Car
+
+MANUFACTURER_URL = reverse("taxi:manufacturer-list")
+DRIVER_URL = reverse("taxi:driver-list")
+CAR_URL = reverse("taxi:car-list")
+
+
+class PublicManufacturerTest(TestCase):
+    def test_login_required(self):
+        response = self.client.get(MANUFACTURER_URL)
+        self.assertNotEqual(200, response.status_code)
+
+
+class PrivateManufacturerTest(TestCase):
+    def setUp(self):
+        self.user = get_user_model().objects.create_user(
+            username="bob123",
+            password="test123",
+        )
+        Manufacturer.objects.create(name="Audi", country="Germany")
+        Manufacturer.objects.create(name="Ford", country="USA")
+        self.manufacturers = Manufacturer.objects.all()
+        self.client.force_login(self.user)
+
+    def test_retrieve_manufacturer(self):
+        response = self.client.get(MANUFACTURER_URL)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
```

Comment:
Good, but you have pagination in your views. You could also try searching on different pages to check whether pagination is correct. There is also simmilar pattern ("?page={some page}")